packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -54,24 +54,32 @@ export async function generateSideloadFile(app: OfficeApp, manifest: ManifestInf
     throw new ExpectedError("The manifest contains an unsupported OfficeApp xsi:type.");
   }
 
-  const templatePath = document && document !== "" ? path.resolve(document) : getTemplatePath(app, addInType);
+  const documentWasProvided = document && document !== "";
+  const templatePath = documentWasProvided ? path.resolve(document) : getTemplatePath(app, addInType);
 
   if (!templatePath) {
     throw new ExpectedError(`Sideload is not supported for apptype: ${addInType}.`);
   }
 
   const templateBuffer = await readFileAsync(templatePath);
-  const zip = await jszip.loadAsync(templateBuffer);
-  const webExtensionPath = getWebExtensionPath(app, addInType);
+  const zip: jszip = await jszip.loadAsync(templateBuffer);
+
+  const appName = getOfficeAppName(app);
+  const extension = path.extname(templatePath);
+  const pathToWrite: string = makePathUnique(
+    path.join(os.tmpdir(), `${appName} add-in ${manifest.id}${extension}`),
+    true
+  );
 
+  if (documentWasProvided) {
```

Comment:
Seems like a simpler change would be to just put an if(!documentWasProvided) around the section that manipulates the file instead of having to break things apart.